rust/agama-manager/src/hardware.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use agama_utils::api::manager::HardwareInfo;
+use serde::Deserialize;
+use serde_with::{formats::PreferMany, serde_as, OneOrMany};
+use std::{
+    path::{Path, PathBuf},
+    process::ExitStatus,
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("lshw command failed")]
+    Command { status: ExitStatus, stderr: String },
+    #[error("Failed to parse lshw output")]
```

Comment:
again, here I think will be useful to either include its output ( which would be sadly quite big ) or content of serde Error which I hope can bring a bit of light to it